### PR TITLE
[INLONG-5158][Sort] Sort add metric for kafka source with flink metrics group and audit sdk

### LINKED ISSUE #5158: [Feature] Sort add metric for kafka source with flink metrics group and audit sdk
### Description

[Feature] Sort add metric for kafka source with flink metrics group and audit sdk

### Use case

[Feature] Sort add metric for kafka source with flink metrics group and audit sdk

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EMsnap**
### [INLONG-5158][Sort] Sort add metric for kafka source with flink metrics group and audit sdk

- Fixes #5158 

### Motivation

add metric for kafka source with flink metrics group and audit sdk

### Modifications

add metric for kafka source with flink metrics group and audit sdk

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/GroupInfo.java [hunk 1]
```diff
@@ -39,6 +40,9 @@ public class GroupInfo implements Serializable {
     @JsonProperty("streams")
     private List<StreamInfo> streams;
 
+    @JsonProperty("extraInfo")
+    private Properties extraInfo;
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/GroupInfo.java:44]: Maybe Properties extraInfo -> Map<String,String> properties ?

Comment (EMsnap) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/GroupInfo.java:44]: thanks for your advice

File: inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        outputMetricForFlink(record);
+        outputMetricForAudit(record);
+    }
+
+    private void outputMetricForAudit(ConsumerRecord<byte[], byte[]> record) {
+        if (auditImp != null) {
+            auditImp.add(
+                Constants.AUDIT_SORT_INPUT,
+                inLongGroupId,
+                inLongStreamId,
```
Comment (gong) [inline inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java:192]: If inlong metric is null, here inlongGroupId and inlongStreamId is null

**Review by yunqingmoswu — APPROVED** [review]
LGTM

**Comment by gong** [comment]
I suggest that expose setting of auditing. For example, send data frequency

**Review by Oneal65 — COMMENTED** [review]
KafkaExtractNode connector is not changed. It's just for kafka-inlong

**Review by gong — APPROVED** [review]

**Review by Oneal65 — APPROVED** [review]

**Review by thesumery — APPROVED** [review]

**Pull request merged**